light-node/src/commands/start.rs

```diff
@@ -108,53 +120,59 @@ impl config::Override<LightNodeConfig> for StartCmd {
-        if !self.rpc_addr.is_empty() {
-            config.rpc_address = self.rpc_addr.to_owned();
+        // Todo figure out if other options would be reasonable to overwrite via CLI arguments.
+        if let Some(addr) = self.listen_addr {
+            config.rpc_config.listen_addr = addr;
         }
-
         Ok(config)
     }
 }
+impl StartCmd {
+    fn make_instance(
+        &self,
+        light_config: &LightClientConfig,
+        io: ProdIo,
+        options: light_client::Options,
+    ) -> Instance {
+        let peer_id = light_config.peer_id;
+        let db_path = light_config.db_path.clone();
 
-/*
- * The following is initialization logic that should have a
- * function in the lite crate like:
- * `subjective_init(height, vals_hash, store, requester) -> Result<(), Error`
- * it would fetch the initial header/vals from the requester and populate a
- * trusted state and store it in the store ...
- * TODO: this should take traits ... but how to deal with the State ?
- * TODO: better name ?
- */
-async fn subjective_init(
-    height: Height,
-    vals_hash: Hash,
-    store: &mut MemStore,
-    req: &RPCRequester,
-) -> Result<(), Error> {
-    if store.get(height).is_ok() {
-        // we already have this !
-        return Ok(());
-    }
-
-    // check that the val hash matches
-    let vals = req.validator_set(height).await?;
+        let db = sled::open(db_path).unwrap_or_else(|e| {
+            println!("[ error ] could not open database: {}", e);
+            std::process::exit(1);
+        });
 
-    if vals.hash() != vals_hash {
-        // TODO
-        panic!("vals hash dont match")
-    }
+        let light_store = SledStore::new(db);
 
-    let signed_header = req.signed_header(height).await?;
+        let state = State {
+            light_store: Box::new(light_store),
+            verification_trace: HashMap::new(),
+        };
 
-    // TODO: validate signed_header.commit() with the vals ...
+        let verifier = ProdVerifier::default();
+        let clock = SystemClock;
+        let scheduler = scheduler::basic_bisecting_schedule;
 
-    let next_vals = req.validator_set(height + 1).await?;
+        let light_client = LightClient::new(peer_id, options, clock, scheduler, verifier, io);
 
-    // TODO: check next_vals ...
+        Instance::new(light_client, state)
+    }
+}
 
-    let trusted_state = &State::new(signed_header, next_vals);
+impl StartCmd {
+    fn assert_init_was_run() {
+        // TODO: handle errors properly:
+        let primary_db_path = app_config().light_clients.first().unwrap().db_path.clone();
+        let db = sled::open(primary_db_path).unwrap_or_else(|e| {
+            println!("[error] could not open database: {}", e);
+            std::process::exit(1);
+        });
 
-    store.add(trusted_state.to_owned())?;
+        let primary_store = SledStore::new(db);
 
-    Ok(())
+        if primary_store.latest(Status::Verified).is_none() {
```

Comment:
Why don't we assert that the latest is `Status::Trusted`?

Reply:
Good question. I thought a lightblock is only considered trusted only if it is supported by witnesses. On init there is only one instance. Need to run this by @milosevic and check the spec.

Reply:
The initial lightblock can be considered as a special case as it is considered trusted by the user. The potential problem I see is that there is an invariant that a trusted block is confirmed with all witnesses. What if initial light block turns out not to be equal to what witnesses consider lightblock for that height? Smells like potential source of bugs.

Reply:
We should just check the initial light block with all witnesses, no?

Reply:
So, we should actually check the initial light block with every peer / witness? Seems a bit weird bc if the header hash matches what is the attack we are trying to prevent by checking it with all witnesses? What would be the consequence. The human component was fooled (subjectively)?